recipes/hdf5/all/conanfile.py

```diff
@@ -1,142 +1,173 @@
 import glob
 import os
 
-from conans import CMake, ConanFile, tools
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
 
-
-class LibHdf5Conan(ConanFile):
+class Hdf5Conan(ConanFile):
     name = "hdf5"
-    description = "HDF5 is a data model, library, and file format for storing and managing data"
-    topics = "conan", "hdf5"
+    description = "HDF5 is a data model, library, and file format for storing and managing data."
+    license = "BSD-3-Clause"
+    topics = ("conan", "hdf5", "hdf", "data")
+    homepage = "https://portal.hdfgroup.org/display/HDF5/HDF5"
     url = "https://github.com/conan-io/conan-center-index"
-    homepage = "https://portal.hdfgroup.org/display/support"
-    license = "MIT"
-    exports_sources = ["CMakeLists.txt"]
+    exports_sources = ["CMakeLists.txt", "patches/**"]
     generators = "cmake"
-
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "enable_cxx": [True, False],
         "hl": [True, False],
-        "with_zlib": [True, False]
+        "threadsafe": [True, False],
+        "with_zlib": [True, False],
+        "szip_support": [None, "with_libaec", "with_szip"],
+        "szip_encoding": [True, False]
```

Comment:
``"szip_support": "with_szip"`` wouldn't mean ``"szip_encoding": True``? I mean, does the ``"szip_encoding"`` option make sense?

Reply:
`"szip_support": "with_szip"` means that we use szip library for szip support.
`"szip_support": "with_libaec"` means that we use libaec library for szip support.

`"szip_encoding"` means that we disable or enable szip encoding in hdf5. For szip library, we have also to ensure that `szip:enable_encoding` is True, because it is not a free feature in szip lib (but it's free in libaec; and libaec recipe doesn't have this option, it's always available).

Reply:
So, in the end it has to be enabled always unless ``"szip_support": None``, right? Why not just adding:

```
def configure(self):
    if self.options.szip_support == "with_szip":
        self.options["szip"].enable_encoding = True
```

Reply:
No, because you may want to only enable decoding. It's fine to use szip (or libaec) library in hdf5 without encoding support.
And I thought it was bad practice to override options or settings.

Reply:
ok, thanks for the clarification.

It is a bad option to override settings/options in the recipe, but here we are just requiring the dependency with a specific option so this is legit